sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs

```diff
@@ -68,17 +73,33 @@ internal async ValueTask<int> AddBatchActivityAsync(Batch<Activity> batchActivit
 
             if (async)
             {
-                response = await this.serviceRestClient.TrackAsync(telemetryItems, cancellationToken).ConfigureAwait(false);
+                response = await this.OnTrackAsync(telemetryItems, cancellationToken).ConfigureAwait(false);
             }
             else
             {
-                response = this.serviceRestClient.TrackAsync(telemetryItems, cancellationToken).Result;
+                response = this.OnTrackAsync(telemetryItems, cancellationToken).Result;
             }
 
             // TODO: Handle exception, check telemetryItems has items
-            return response.Value.ItemsAccepted.GetValueOrDefault();
+            return response?.Value.ItemsAccepted.GetValueOrDefault() ?? default(int);
         }
 
+        /// <summary>
+        /// In a normal configuration, this is expected to invoke the serviceRestClient and transmit telemetry to the ingestion service.
+        /// This property can be changed to override the behavior of transmitting telemetry.
+        /// This can be used for Unit Testing to capture telemetry that would have been transmitted.
+        /// </summary>
+        internal Func<IEnumerable<TelemetryItem>, CancellationToken, Task<Response<TrackResponse>>> OnTrackAsync { private get; set; }
```

Comment:
Why do we need this at all?

Reply:
I'm open to alternate approaches. 
The very last action the Exporter/Transmitter execute is to send telemetry to the ServiceRestClient.
I want the unit tests to be able change this behavior and say, send telemetry to me instead.

- The goal here is to allow telemetry to pass fully through the Exporter. 
- I want the actual TelemetryItem object so we're not trying to capture http requests or serialize Json within a test.

Reply:
Consider mock the underlying ServiceRestClient.

Reply:
I thought of that but the current `ServiceRestClient` is internal. 
I could instead introduce a new interface; `IServiceRestClient` and keep the current implementation internal.

Would you approve of that?

Reply:
> I thought of that but the current `ServiceRestClient` is internal.

What's the blocking issue with "the current `ServiceRestClient` is internal"?

Reply:
I want to be able to provide a Public API for unit testing.
My IntegrationTest app currently has no access to internal classes, so I won't be able to mock the ServiceRestClient using a tool like Moq. 
Maybe I'm not following if you have something else in mind.

Reply:
This is what's in my mind:
* We don't want to expose public API "just for our own testing purpose".
* Test cases should have access to internal members of the target.